cdap-ui/app/cdap/components/PreviewData/index.tsx

```diff
@@ -66,10 +72,22 @@ const PreviewDataViewBase: React.FC<IPreviewDataViewProps> = ({
 
   const [previewLoading, setPreviewLoading] = useState(false);
   const [previewData, setPreviewData] = useState<IPreviewData>({});
+  const [tableData, setTableData] = useState<ITableData>({});
   const [error, setError] = useState(null);
+  const [viewMode, setViewMode] = useState('Table');
+
+  const widgetProps = {
+    on: { value: 'Record', label: 'Record' },
```

Comment:
Maybe we can define `Record` and `Table` with enum, since we use them more than once in this `PreviewData/index.tsx`.

Reply:
Good idea! Done.